specification/hdinsight/resource-manager/Microsoft.HDInsight/HDInsightOnAks/preview/2023-11-01-preview/examples/CreateClusterPoolWithPrivateAks.json

```diff
@@ -0,0 +1,107 @@
+          "clusterPoolProfile": {
+            "clusterPoolVersion": "1.2"
+          },
+          "computeProfile": {
+            "vmSize": "Standard_D3_v2",
+            "count": 3
+          },
+          "aksClusterProfile": {
+            "aksVersion": "1.24",
+            "aksClusterAgentPoolIdentityProfile": {
+              "msiResourceId": "/subscriptions/subid/resourceGroups/hdi-45cd32aead6e4a91b079a0cdbfac8c36/providers/Microsoft.ManagedIdentity/userAssignedIdentities/clusterpool1-agentpool",
+              "msiClientId": "a89fb478-2a84-4d9b-8f18-3e8c4d1db3eb",
+              "msiObjectId": "dc7ef861-8b55-4ffb-9003-20885cd895a9"
+            },
+            "aksClusterResourceId": "/subscriptions/subid/resourceGroups/hdi-45cd32aead6e4a91b079a0cdbfac8c36/providers/Microsoft.ContainerService/managedClusters/clusterpool1"
+          },
+          "networkProfile": {
+            "subnetId": "/subscriptions/subid/resourceGroups/hiloResourcegroup/providers/Microsoft.Network/virtualNetworks/vnet1/subnets/subnet1",
+            "enablePrivateApiServer": true
+          }
+        },
+        "systemData": {
+          "createdBy": "string",
+          "createdByType": "User",
+          "createdAt": "2020-02-03T01:01:01.1075056Z",
+          "lastModifiedBy": "string",
+          "lastModifiedByType": "User",
+          "lastModifiedAt": "2020-02-04T02:03:01.1974346Z"
+        },
+        "location": "West US 2"
+      }
+    },
+    "201": {
+      "body": {
+        "id": "/subscriptions/subid/resourceGroups/hiloResourcegroup/providers/Microsoft.HDInsight/clusterPools/clusterpool1",
+        "name": "clusterpool1",
+        "type": "Microsoft.HDInsight/clusterPools",
+        "properties": {
+          "provisioningState": "Accepted",
+          "deploymentId": "45cd32aead6e4a91b079a0cdbfac8c36",
+          "managedResourceGroupName": "hdi-45cd32aead6e4a91b079a0cdbfac8c36",
+          "clusterPoolProfile": {
+            "clusterPoolVersion": "1.2"
+          },
+          "computeProfile": {
+            "vmSize": "Standard_D3_v2",
+            "count": 3
+          },
+          "aksClusterProfile": {
+            "aksVersion": "1.24",
+            "aksClusterAgentPoolIdentityProfile": {
+              "msiResourceId": "/subscriptions/subid/resourceGroups/hdi-45cd32aead6e4a91b079a0cdbfac8c36/providers/Microsoft.ManagedIdentity/userAssignedIdentities/clusterpool1-agentpool",
+              "msiClientId": "a89fb478-2a84-4d9b-8f18-3e8c4d1db3eb",
+              "msiObjectId": "dc7ef861-8b55-4ffb-9003-20885cd895a9"
+            },
+            "aksClusterResourceId": "/subscriptions/subid/resourceGroups/hdi-45cd32aead6e4a91b079a0cdbfac8c36/providers/Microsoft.ContainerService/managedClusters/clusterpool1"
+          },
+          "networkProfile": {
+            "subnetId": "/subscriptions/subid/resourceGroups/hiloResourcegroup/providers/Microsoft.Network/virtualNetworks/vnet1/subnets/subnet1",
+            "enablePrivateApiServer": true
```

Comment:
>"enablePrivateApiServer": true [](http://example.com/codeflow?start=12&length=30)

could consider instead modeling this as e.g.
"apiServerEndpointType": "private" | "public"

Reply:
After discussion, our team prefer to use boolean because it's more easy to be understanded by customer in our scenario.